src/SignhostAPIClient/Rest/SignHostApiClient.cs

```diff
@@ -26,6 +26,52 @@ public class SignHostApiClient
 		private readonly ISignHostApiClientSettings settings;
 		private readonly HttpClient client;
 
+		public static bool IsPostbackChecksumValid(
+			IDictionary<string, string[]> headers,
+			string body,
+			string sharedSecret,
+			out Transaction postbackTransaction)
+		{
+			postbackTransaction = null;
+			string[] postbackChecksumArray;
+			string postbackChecksum;
+			string calculatedChecksum;
+			Postback postback;
+
+			try {
+				postback = JsonConvert.DeserializeObject<Postback>(body);
+				postbackTransaction = postback;
+			} catch (InvalidCastException e) {
+				return false;
+			} catch {
+				return false;
+			}
+
+			if (headers.ContainsKey("Checksum")) {
+				headers.TryGetValue("Checksum", out postbackChecksumArray);
+				postbackChecksum = postbackChecksumArray.First();
+			} else {
+				postbackChecksum = postback.Checksum;
+			}
+
+			if (!string.IsNullOrEmpty(postbackTransaction.Id) &&
+				!string.IsNullOrEmpty(postbackChecksum)) {
+				using (var sha1 = SHA1.Create()) {
+					var preCalculatedChecksum = sha1.ComputeHash(Encoding.UTF8.GetBytes(
+						$"{postbackTransaction.Id}||" +
+						$"{(int)postbackTransaction.Status}|" +
+						$"{sharedSecret}"));
+					calculatedChecksum = BitConverter.ToString(
+						preCalculatedChecksum)
+ 						.Replace("-", string.Empty)
+						.ToLower();
+				}
+			} else {
+				return false;
+			}
+			return Equals(calculatedChecksum, postbackChecksum);
```

Comment:
Always leave a newline after a closing `}` unless it is immediatly followed by another `}`